### PR TITLE
Added Result#asList method

### PR DESCRIPTION
**Pull request by ernstsson**

### CODE REVIEW

File: agera/src/main/java/com/google/android/agera/Result.java [hunk 1]
```diff
@@ -161,6 +166,27 @@ public T get() throws FailedResultException {
     throw new FailedResultException(failure);
   }
 
+
+  /**
+   * Returns a list containing the value if it is present, or an empty list.
+   */
+  @NonNull
+  public List<T> asList() {
+    if (value == null) {
```
Comment (maxtroy) [inline agera/src/main/java/com/google/android/agera/Result.java]: actually I have a slight optimization idea: immediately set `list` to `emptyList()` if Result is constructed with a Throwable. emptyList() is a system-level constant so it doesn't take space (amortized cost); and whether an object field is null or non-null doesn't affect the memory used.

File: agera/src/test/java/com/google/android/agera/ResultTest.java [hunk 2]
```diff
@@ -685,8 +687,8 @@ public void shouldBeSingletonForFailureWithoutExplicitThrowable() {
   }
 
   @Test
-  public void shouldVerifyEqualsForSqlDeleteRequest() {
-    EqualsVerifier.forClass(Result.class).verify();
+  public void shouldVerifyEqualsForResult() {
+    EqualsVerifier.forClass(Result.class).withIgnoredFields("list").verify();
```
Comment (maxtroy) [inline agera/src/test/java/com/google/android/agera/ResultTest.java]: Try `transient` on `list`:

http://docs.oracle.com/javase/specs/jls/se7/html/jls-8.html#jls-8.3.1.3

http://jqno.nl/equalsverifier/manual/ignoring-fields/ (search 'transient')

**Comment by ernstsson** [comment]
ok, doing the new commit thing instead then, let me know if this works better :)

File: agera/src/test/java/com/google/android/agera/ResultTest.java [hunk 3]
```diff
@@ -688,7 +688,7 @@ public void shouldBeSingletonForFailureWithoutExplicitThrowable() {
 
   @Test
   public void shouldVerifyEqualsForResult() {
-    EqualsVerifier.forClass(Result.class).withIgnoredFields("list").verify();
+    forClass(Result.class).verify();
```
Comment (maxtroy) [inline agera/src/test/java/com/google/android/agera/ResultTest.java]: static import here makes code less readable :)

Comment (ernstsson) [inline agera/src/test/java/com/google/android/agera/ResultTest.java]: yeah, makes sense

**Pull request merged**